spring-context-support/src/main/java/org/springframework/cache/transaction/AbstractTransactionSupportingCacheManager.java

```diff
@@ -16,46 +16,36 @@
 
 package org.springframework.cache.transaction;
 
-import org.springframework.cache.Cache;
 import org.springframework.cache.support.AbstractCacheManager;
 
 /**
  * Base class for CacheManager implementations that want to support built-in
  * awareness of Spring-managed transactions. This usually needs to be switched
- * on explicitly through the {@link #setTransactionAware} bean property.
+ * on explicitly through the {@link TransactionAwareCacheDecorator#transactionAware} bean property.
  *
  * @author Juergen Hoeller
- * @since 3.2
- * @see #setTransactionAware
- * @see TransactionAwareCacheDecorator
+ * @see TransactionAwareCache
  * @see TransactionAwareCacheManagerProxy
+ * @since 3.2
  */
 public abstract class AbstractTransactionSupportingCacheManager extends AbstractCacheManager {
 
-	private boolean transactionAware = false;
-
-
 	/**
-	 * Set whether this CacheManager should expose transaction-aware Cache objects.
-	 * <p>Default is "false". Set this to "true" to synchronize cache put/evict
-	 * operations with ongoing Spring-managed transactions, performing the actual cache
-	 * put/evict operation only in the after-commit phase of a successful transaction.
+	 * Constructor that creates CacheManager with transactionAware flag set to false.
 	 */
-	public void setTransactionAware(boolean transactionAware) {
-		this.transactionAware = transactionAware;
+	public AbstractTransactionSupportingCacheManager() {
+		this(false);
 	}
 
 	/**
-	 * Return whether this CacheManager has been configured to be transaction-aware.
+	 * Constructor that creates CacheManager with the given transactionAware flag.
+	 *
+	 * @param transactionAware Set whether this CacheManager should expose transaction-aware Cache objects.
+	 *                         <p>Default is "false". Set this to "true" to synchronize cache put/evict
+	 *                         operations with ongoing Spring-managed transactions, performing the actual cache
+	 *                         put/evict operation only in the after-commit phase of a successful transaction.
 	 */
-	public boolean isTransactionAware() {
```

Comment:
I don't see a reason why introducing a concept of `CacheDecorator` should remove those methods. IMO, that should be left untouched (same remark as my first review). This also means that the new `TransactionAwareCacheDecorator` in your proposal should probably be an private concern of this class (rather than a public class).